nevergrad/optimization/optimizerlib.py

```diff
@@ -1815,6 +1816,196 @@ def __init__(
 )
 
 
+class _PCABO(base.Optimizer):
+
+    def __init__(
+            self,
+            parametrization: IntOrParameter,
+            budget: tp.Optional[int] = None,
+            num_workers: int = 1,
+            *,
+            init_budget: tp.Optional[int] = None,
+            n_components: tp.Optional[float] = 0.95,
+            prop_doe_factor: tp.Optional[float] = None,
+    ) -> None:
+        super().__init__(parametrization, budget=budget, num_workers=num_workers)
+
+        from bayes_optim.extension import PCABO as PcaBO
+        from bayes_optim.extension import RealSpace
+
+        lb, ub = 1e-7 - np.pi / 2, np.pi / 2 - 1e-7
+        space = RealSpace([lb, ub]) * self.dimension
+        self.buffer: tp.List[float] = []
+        self.newX: tp.List[float] = []
+        self.loss: tp.List[float] = []
+
+        # Setting DoE size as a percentage of the total budget if prop_doe_factor is not None
+        if prop_doe_factor:
+            init_budget = prop_doe_factor * budget
+
+        self._pcabo = PcaBO(
+            search_space=space,
+            obj_fun=None,  # Assuming that this is not used :-)
+            DoE_size=init_budget if init_budget is not None else 5,
+            max_FEs=budget,
+            verbose=True,
+            n_point=1,  # We start with a sequential procedure, maybe we'll extend in a second moment
+            n_components=n_components,
+            acquisition_optimization={"optimizer": "BFGS"},
+        )
+
+    def _internal_ask_candidate(self) -> p.Parameter:
+        if not self.buffer:
+            self.newX = []
+            self.loss = []
```

Comment:
let's "hide" it with an underscore, and make it a plural for clarity: `self._losses`

Reply:
Done